src/main/java/bio/terra/cloudfunctions/common/FileTypeDetector.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.cloudfunctions.common;
```

Comment:
I think we can rely on storageObjectData.getContentType() for now.  I don't think we need this class.

Reply:
The `FileTypeDetector` is intended to gather sufficient information to decide which concrete implementation of the Cloud Function to invoke. I echoed @davidangb comments 9 days ago on this topic - `Content type` is in general not sufficient unless we take Option 2 below, quote:

`I think that simply looking at the Content-Type of the incoming file is restrictive and may cause us headaches in the future. For instance, what if QA and App Services both want to process some json, but using different json schemas? Both of these would show a content type of application/json, but they would need different handlers.

I'd prefer one of two routes:

1. give as much information to the switching logic as possible - this would include the file path and bucket, the content-type, the triggering event type, really anything we can think of. Then, developers can write flexible switching logic such as "if it is json, inside the 'foo' bucket, and with an '.upsert' extension, use class Y for processing".

2. specify the handler class in a config file. App Services, QA, and future callers can deploy their instance and specify their main handler class in config; at runtime, we'll simply use that handler class and avoid switching logic. If a given deployment needs custom switching logic, they can write it themselves in their handler class.`

Either option is fine with me.